### DISCUSSION TITLE
Is it possible to change the system time that bullmq uses to run delayed jobs?

### DISCUSSION THREAD

**Discussion by stasgm** [taskforcesh/bullmq#2382] [Q&A]
When testing it is very useful to be able to control time. We have a module that can change the 'internal clock' of the server and emulate the time to be in present or future. This is very helpful when testing if certain action that needs to happen X minutes/hours/days from now will actually happen. 

Is it possible to change the system time inside Bullmq when running the server?

**Comment by manast** [comment]
We use faketimers in our tests, maybe this is something you can look into: https://www.npmjs.com/package/@sinonjs/fake-timers
